app/assets/javascripts/discourse/app/lib/minimasonry.js

```diff
@@ -0,0 +1,29 @@
+import loadScript from "discourse/lib/load-script";
+
+export default function (elem, site) {
+  if (!elem) {
+    return;
+  }
+
+  const imageGrids = elem.querySelectorAll(
+    '.auto-image-grid[data-auto-image-grid="on"]'
+  );
+
+  if (!imageGrids.length) {
+    return;
+  }
+
+  loadScript("/javascripts/minimasonry.min.js").then(function () {
```

Comment:
This is gonna make the layout happen asynchronously, right? We need to be careful about 'jumpy height' in topics.

If we want to use this library, I think we need to include it in our main JS bundle (i.e. add it to our `package.json`). Or alternatively, we do the masonry layout ourselves. Given that the images have known aspect ratios, it's a fairly simple calculation. You may be interested in the Masonry implementation we have in [discourse-topic-thumbnails](https://github.com/discourse/discourse-topic-thumbnails/blob/bd495f7aca92bcb4e61a199d2a28c974110b12d1/javascripts/discourse/initializers/topic-thumbnails-init.js#L137-L171)